Assets/MixedRealityToolkit/MSBuild/Scripts/CSProjectInfo.cs

```diff
@@ -0,0 +1,363 @@
+            }
+            sourceGuidToClassName.Add(guid, classNameToAdd);
+
+
+            string normalized = Utilities.GetNormalizedPath(sourceFile);
+            if (normalized.StartsWith("Packages"))
+            {
+                normalized = "PackagesCopy" + normalized.Substring("Packages".Length);
+            }
+            sourceIncludes.Add(Utilities.ReplaceTokens(sourceIncludeTemplate, new Dictionary<string, string>()
+            {
+                {"##RELATIVE_SOURCE_PATH##", $"..\\{normalized}" },
+                {"##PROJECT_LINK_PATH##", normalized.Replace("Assets\\", string.Empty) }
+            }));
+        }
+
+        private void PopulateSupportedPlatformBuildConditions(List<string> supportedPlatformBuildConditions, string template, string configuration, IReadOnlyDictionary<BuildTarget, CompilationPlatformInfo> platforms)
+        {
+            foreach (KeyValuePair<BuildTarget, CompilationPlatformInfo> platform in platforms)
+            {
+                supportedPlatformBuildConditions.Add(Utilities.ReplaceTokens(template, new Dictionary<string, string>()
+                {
+                    { "##SUPPORTED_CONFIGURATION_TOKEN##", configuration },
+                    { "##SUPPORTED_PLATFORM_TOKEN##", platform.Value.Name }
+                }));
+            }
+        }
+
+        private string CreateProjectReferencesSet(string template, HashSet<string> additionalSearchPaths, bool inEditor)
+        {
+            if (Utilities.TryGetXMLTemplate(template, "PROJECT_REFERENCE", out string projectReferenceTemplate)
+                && Utilities.TryGetXMLTemplate(template, "PLUGIN_REFERENCE", out string pluginReferenceTemplate))
+            {
+                List<string> projectReferences = new List<string>();
+                foreach (CSProjectDependency<CSProjectInfo> dependency in csProjectDependencies)
+                {
+                    List<string> platformConditions = GetPlatformConditions(inEditor ? InEditorPlatforms : PlayerPlatforms, inEditor ? dependency.InEditorSupportedPlatforms : dependency.PlayerSupportedPlatforms);
+
+                    projectReferences.Add(Utilities.ReplaceTokens(projectReferenceTemplate, new Dictionary<string, string>()
+                    {
+                        { "##REFERENCE_TOKEN##", $"{dependency.Dependency.Name}.csproj" },
+                        { "<!--HINT_PATH_TOKEN-->", dependency.Dependency.ReferencePath.AbsolutePath },
+                        { "##CONDITION_TOKEN##", platformConditions.Count == 0 ? "false" : string.Join(" OR ", platformConditions)}
+                    }));
+                }
+
+                List<string> pluginReferences = new List<string>();
+                foreach (CSProjectDependency<PluginAssemblyInfo> dependency in pluginAssemblyDependencies)
+                {
+                    if (dependency.Dependency.Type == PluginType.Native)
+                    {
+                        //TODO Native plugins aren't yet supported
```

Comment:
>//TODO Native plugins aren't yet supported [](start = 24, length = 42)

We don't like to have blank TODOs checked in here - if you can add more details or have an issue filed (and then link to that issue) that would be best. Otherwise this will probably live on forever with little background